libvirt/network.go

```diff
@@ -166,31 +38,54 @@ func waitForNetworkDestroyed(virConn *libvirt.Connect, uuid string) resource.Sta
 	}
 }
 
-func setDhcpByCIDRAdressesSubnets(d *schema.ResourceData, networkDef *libvirtxml.Network) error {
-	if addresses, ok := d.GetOk("addresses"); ok {
-		ipsPtrsLst := []libvirtxml.NetworkIP{}
-		for _, addressI := range addresses.([]interface{}) {
-			// get the IP address entry for this subnet (with a guessed DHCP range)
-			dni, dhcp, err := setNetworkIP(addressI.(string))
-			if err != nil {
-				return err
-			}
-			if d.Get("dhcp.0.enabled").(bool) {
-				dni.DHCP = dhcp
-			} else {
-				// if a network exist with enabled but an user want to disable
-				// dhcp, we need to set dhcp struct to nil.
-				dni.DHCP = nil
-			}
-
-			ipsPtrsLst = append(ipsPtrsLst, *dni)
+// getNetModeFromResource returns the network mode fromm a network definition
+func getNetModeFromResource(d *schema.ResourceData) string {
+	return strings.ToLower(d.Get("mode").(string))
+}
+
+// getIPsFromResource gets the IPs configurations from the resource definition
+func getIPsFromResource(d *schema.ResourceData) ([]libvirtxml.NetworkIP, error) {
+	addresses, found := d.GetOk("addresses")
+	if !found {
```

Comment:
`nitpick`. We might use here instead of `found`  `ok` var, since that is used in the whole codebase and we are more coherent to that.

Reply:
(this for all `found`)